include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -115,82 +109,92 @@ struct WorkSpace {
   //
   //        post_sorted <--- matmul
   //
-  Matrix<SizeType, Device::CPU> i1;
-  Matrix<SizeType, Device::CPU> i2;
-  Matrix<SizeType, Device::CPU> i3;
+  Matrix<SizeType, device> i1;
+  Matrix<SizeType, device> i2;
+  Matrix<SizeType, device> i3;
 
   // Assigns a type to each column of Q which is used to calculate the permutation indices for Q and U
   // that bring them in matrix multiplication form.
-  Matrix<ColType, Device::CPU> c;
+  Matrix<ColType, device> c;
+};
+
+// Device::GPU
+template <class T, Device D>
+struct WorkSpaceHostMirror {
+  matrix::Matrix<T, Device::CPU> evals;
+  matrix::Matrix<T, Device::CPU> mat1;
+  matrix::Matrix<T, Device::CPU> dtmp;
+  matrix::Matrix<T, Device::CPU> z;
+  matrix::Matrix<T, Device::CPU> ztmp;
+  matrix::Matrix<SizeType, Device::CPU> i2;
+  matrix::Matrix<ColType, Device::CPU> c;
+};
+
+template <class T>
+struct WorkSpaceHostMirror<T, Device::CPU> {
+  matrix::Matrix<T, Device::CPU>& evals;
+  matrix::Matrix<T, Device::CPU>& mat1;
+  matrix::Matrix<T, Device::CPU>& dtmp;
+  matrix::Matrix<T, Device::CPU>& z;
+  matrix::Matrix<T, Device::CPU>& ztmp;
+  matrix::Matrix<SizeType, Device::CPU>& i2;
+  matrix::Matrix<ColType, Device::CPU>& c;
```

Comment:
Is this a duplication? Is this needed?

Reply:
When on CPU, these just serve as references to the original data. This is similar to how MatrixMirror works, except that I can use these to copy subranges of tiles for each matrix and there is no copy upon destruction. Note that no copy is performed if two references point to the same object as is the case on CPU. The copies only happen when executing on the GPU.